esp-hal/ld/sections/stack.x

```diff
@@ -1,43 +1,27 @@
-#IF ESP_HAL_CONFIG_FLIP_LINK
-/* no Xtensa chip is supported - so we can assume RISC-V */
 SECTIONS {
   /* must be last segment using RWDATA */
   .stack (NOLOAD) : ALIGN(4)
   {
     _stack_end = ABSOLUTE(.);
     _stack_end_cpu0 = ABSOLUTE(.);
 
+    /*
+    Provide the stack_guard for `stack-protector`
+
+    TODO: Ideally the offset should be configurable - should be done once we have https://github.com/esp-rs/esp-hal/issues/1111
```

Comment:
I've just opened https://github.com/esp-rs/esp-hal/issues/3213 could you replace this comment with a link to here?

Reply:
I'd rather spend the 5 minutes fixing the thing altogether if someone can tell me what the symbol is actually used for.

Reply:
it's this: https://doc.rust-lang.org/rustc/exploit-mitigations.html#stack-smashing-protection

Reply:
and by having the symbol at the bottom of the stack it also triggers when something overwrites the value